crates/floresta-chain/src/pruned_utreexo/consensus.rs

```diff
@@ -208,29 +214,22 @@ impl Consensus {
         }
         Ok(())
     }
-    /// Consumes the UTXOs from the hashmap, and returns the value of the consumed UTXOs.
-    /// If we do not find the UTXO, we return an error invalidating the input that tried to
-    /// consume that UTXO.
-    fn consume_utxos(
+    /// Returns the TxOut being spent by the given input.
+    ///
+    /// Fails if the UTXO is not present in the given hashmap.
+    fn get_utxo<'a>(
         input: &TxIn,
-        utxos: &mut HashMap<OutPoint, TxOut>,
-        value_var: &mut u64,
-    ) -> Result<(), BlockValidationErrors> {
+        utxos: &'a HashMap<OutPoint, TxOut>,
+    ) -> Result<&'a TxOut, BlockValidationErrors> {
         match utxos.get(&input.previous_output) {
-            Some(prevout) => {
-                *value_var += prevout.value.to_sat();
-                utxos.remove(&input.previous_output);
-            }
-            None => {
-                return Err(BlockValidationErrors::UtxoAlreadySpent(
-                    //This is the case when the spender:
-                    // - Spends an UTXO that doesn't exist
-                    // - Spends an UTXO that was already spent
-                    input.previous_output.txid,
-                ));
-            }
-        };
-        Ok(())
+            Some(txout) => Ok(txout),
+            None => Err(
+                // This is the case when the spender:
+                // - Spends an UTXO that doesn't exist
+                // - Spends an UTXO that was already spent
+                BlockValidationErrors::UtxoAlreadySpent(input.previous_output.txid),
+            ),
```

Comment:
Maybe this variant of the `BlockValidationErrors` could be renamed to `UtxoNotFound`.

Because this error occurs when the UTXO is not found... The supposition that is already spent can lead to misunderstanding.

Reply:
Agree